evidence/pool.go

```diff
@@ -300,60 +517,66 @@ func (evpool *Pool) listEvidence(prefixKey byte, maxNum int64) ([]types.Evidence
 
 		val := iter.Value()
 		var (
-			ev   types.Evidence
-			evpb tmproto.Evidence
+			evInfo Info
+			evpb   evproto.Info
 		)
-		err := proto.Unmarshal(val, &evpb)
+		err := evpb.Unmarshal(val)
 		if err != nil {
 			return nil, err
 		}
 
-		ev, err = types.EvidenceFromProto(&evpb)
+		evInfo, err = InfoFromProto(&evpb)
 		if err != nil {
 			return nil, err
 		}
 
-		evidence = append(evidence, ev)
+		evidence = append(evidence, evInfo.Evidence)
 	}
 
 	return evidence, nil
 }
 
-func (evpool *Pool) removeExpiredPendingEvidence() {
+func (evpool *Pool) removeExpiredPendingEvidence() (int64, time.Time) {
 	iter, err := dbm.IteratePrefix(evpool.evidenceStore, []byte{baseKeyPending})
 	if err != nil {
 		evpool.logger.Error("Unable to iterate over pending evidence", "err", err)
-		return
+		return evpool.State().LastBlockHeight, evpool.State().LastBlockTime
 	}
 	defer iter.Close()
 	blockEvidenceMap := make(map[string]struct{})
 	for ; iter.Valid(); iter.Next() {
 		evBytes := iter.Value()
 		var (
-			ev   types.Evidence
-			evpb tmproto.Evidence
+			evInfo   Info
+			evInfoPb evproto.Info
 		)
-		err := proto.Unmarshal(evBytes, &evpb)
+		err := evInfoPb.Unmarshal(evBytes)
```

Comment:
can we extract this into a function? I think I've seen it before (**duplication**)